### PR TITLE
Add device and sycl_queue keyword arguments to random calls

### PR DESCRIPTION
**Pull request by antonwolfy**
This PR is aimed to implement `device` and `sycl_queue` keyword arguments in a part of legacy functions from `dpnp.random` module to enable a compliance with Compute Follows Data programming model for them.

The full list of impacted functions is
- dpnp.random.normal()
- dpnp.random.rand()
- dpnp.random.randint()
- dpnp.random.randn()
- dpnp.random.random()
- dpnp.random.random_integers()
- dpnp.random.random_sample()
- dpnp.random.ranf()
- dpnp.random.sample()
- dpnp.random.seed()
- dpnp.random.standard_normal()
- dpnp.random.uniform()

The issue with missing keywords was addressed in #1260.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

**Comment by oleksandr-pavlyk** [comment]
Probably not related to change in this commit, but `dpnp.random.randint(100, size=10, device="gpu")` is supposed to work, and translate to `dpnp.random.randint(0, 100, size=10, device="gpu")`.

File: dpnp/random/dpnp_iface_random.py [hunk 1]
```diff
@@ -98,11 +98,18 @@
...
 
 
-def _get_random_state():
-    global _dpnp_random_state
-    if _dpnp_random_state is None:
-        _dpnp_random_state = RandomState()
-    return  _dpnp_random_state
+def _get_random_state(device=None, sycl_queue=None):
+    global _dpnp_random_states
+
```
Comment (oleksandr-pavlyk) [inline dpnp/random/dpnp_iface_random.py:103]:
```suggestion
    global _dpnp_random_states

    if not isinstance(_dpnp_random_states, dict):
         _dpnp_random_states = dict()
```

Ensure that internal variable can not be reset to an object that does not support `__setitem__` with `SyclQueue` as a key.

**Review by oleksandr-pavlyk — APPROVED** [review]
Overall looks good! Thank you @antonwolfy

**Comment by antonwolfy** [comment]
> Probably not related to change in this PR, but `dpnp.random.randint(100, size=10, device="gpu")` is supposed to work, and translate to `dpnp.random.randint(0, 100, size=10, device="gpu")`.

I've created a separate PR #1284 to implement that.

**Review by oleksandr-pavlyk — APPROVED** [review]

**Pull request merged**